src/main/java/com/clickhouse/kafka/connect/sink/db/ClickHouseWriter.java

```diff
@@ -451,9 +464,19 @@ public void doInsertJson(List<Record> records) throws IOException, ExecutionExce
                 // write bytes into the piped stream
                 for (Record record: records ) {
                     if (record.getSinkRecord().value() != null ) {
-                        Map<String, Object> data = (Map<String, Object>) record.getSinkRecord().value();
-                        java.lang.reflect.Type gsonType = new TypeToken<HashMap>() {
-                        }.getType();
+                        Map<String, Object> data;
+                        if (record.getSinkRecord().value() instanceof Map) {
```

Comment:
we can determine that with getSchemaType() instead of instanceof (less preferment)

Reply:
I think switch with 2 cases is cleaner than instanceof